airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/uploader/BigQueryDirectUploader.java

```diff
@@ -4,29 +4,205 @@
+import com.google.cloud.bigquery.Field;
+import com.google.cloud.bigquery.Job;
+import com.google.cloud.bigquery.JobInfo;
 import com.google.cloud.bigquery.JobInfo.WriteDisposition;
+import com.google.cloud.bigquery.QueryJobConfiguration;
+import com.google.cloud.bigquery.Schema;
+import com.google.cloud.bigquery.StandardTableDefinition;
+import com.google.cloud.bigquery.Table;
+import com.google.cloud.bigquery.TableDataWriteChannel;
 import com.google.cloud.bigquery.TableId;
+import com.google.cloud.bigquery.TableInfo;
+import io.airbyte.cdk.integrations.base.AirbyteExceptionHandler;
+import io.airbyte.cdk.integrations.base.JavaBaseConstants;
+import io.airbyte.cdk.integrations.destination.async.model.PartialAirbyteMessage;
+import io.airbyte.commons.string.Strings;
 import io.airbyte.integrations.destination.bigquery.BigQueryUtils;
 import io.airbyte.integrations.destination.bigquery.formatter.BigQueryRecordFormatter;
 import io.airbyte.integrations.destination.bigquery.writer.BigQueryTableWriter;
 import io.airbyte.protocol.models.v0.AirbyteMessage;
+import java.io.IOException;
 import java.util.function.Consumer;
+import java.util.stream.Collectors;
+import org.apache.commons.lang3.tuple.ImmutablePair;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class BigQueryDirectUploader {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(BigQueryDirectUploader.class);
+
+  protected final TableId table;
+  protected final WriteDisposition syncMode;
+  protected final BigQueryTableWriter writer;
+  protected final BigQuery bigQuery;
+  protected final BigQueryRecordFormatter recordFormatter;
+
+  BigQueryDirectUploader(final TableId table,
+                         final BigQueryTableWriter writer,
+                         final WriteDisposition syncMode,
+                         final BigQuery bigQuery,
+                         final BigQueryRecordFormatter recordFormatter) {
+    this.table = table;
+    this.writer = writer;
+    this.syncMode = syncMode;
+    this.bigQuery = bigQuery;
+    this.recordFormatter = recordFormatter;
+  }
+
+  public void upload(final PartialAirbyteMessage airbyteMessage) {
+    try {
+      writer.write(recordFormatter.formatRecord(airbyteMessage));
+    } catch (final IOException | RuntimeException e) {
+      LOGGER.error("Got an error while writing message: {}", e.getMessage(), e);
+      LOGGER.error(String.format(
+          "Failed to process a message for job: %s",
+          writer.toString()));
+      printHeapMemoryConsumption();
+      throw new RuntimeException(e);
+    }
+  }
+
 
-public class BigQueryDirectUploader extends AbstractBigQueryUploader<BigQueryTableWriter> {
+  public void closeAfterPush() {
+    try {
+      this.writer.close();
+    } catch (final IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  public void createRawTable() {
+    // Ensure that this table exists.
+    // TODO alter an existing raw table?
```

Comment:
delete this TODO, if we want to alter existing raw tables that should go into a Migration object